### PR TITLE
Port categories/slugs over to Diesel

### PR DESCRIPTION
**Pull request by sgrif**

### CODE REVIEW

File: src/category.rs [hunk 1]
```diff
@@ -388,29 +388,17 @@ pub fn show(req: &mut Request) -> CargoResult<Response> {
...
 pub fn slugs(req: &mut Request) -> CargoResult<Response> {
-    let conn = req.tx()?;
-    let stmt = conn.prepare(
-        "SELECT slug FROM categories \
-         ORDER BY slug",
-    )?;
-    let rows = stmt.query(&[])?;
+    let conn = req.db_conn()?;
+    let slugs = categories::table
+        .select((categories::slug, categories::slug))
```
Comment (carols10cents) [inline src/category.rs]: I'd still like the order by to ensure that the slugs are alphabetical, please :)

Comment (sgrif) [inline src/category.rs]: Lol wtf why did I write this

**Comment by sgrif** [comment]
Updated and added a test for this endpoint.

**Comment by sgrif** [comment]
Failure was just `rustfmt`. I fixed, going to go ahead and merge

**Pull request merged**